org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/NexusSlit.java

```diff
@@ -0,0 +1,254 @@
+			double fill = "NaN".equalsIgnoreCase(floatFill) ? Double.NaN : Double.parseDouble(floatFill);
+
+			xLzSet = slit.initializeLazyDataset(NXslit.NX_X_GAP, 1, Double.class);
+			yLzSet = slit.initializeLazyDataset(NXslit.NX_Y_GAP, 1, Double.class);
+			xLzSet.setFillValue(fill);
+			yLzSet.setFillValue(fill);
+			xLzSet.setChunking(new int[]{8}); // Faster than looking at the shape of the scan for this dimension because slow to iterate.
+			yLzSet.setChunking(new int[]{8}); // Faster than looking at the shape of the scan for this dimension because slow to iterate.
+			xLzSet.setWritingAsync(true);
+			yLzSet.setWritingAsync(true);
+
+			xLzValue = slit.initializeLazyDataset(NXslit.NX_X_GAP, info.getRank(), Double.class);
+			yLzValue = slit.initializeLazyDataset(NXslit.NX_Y_GAP, info.getRank(), Double.class);
+			xLzValue.setFillValue(fill);
+			yLzValue.setFillValue(fill);
+			xLzValue.setChunking(info.createChunk(false, 8)); // Might be slow, need to check this
+			yLzValue.setChunking(info.createChunk(false, 8)); // Might be slow, need to check this
+			xLzValue.setWritingAsync(true);
+			yLzValue.setWritingAsync(true);
+		}
+
+		registerAttributes(slit, this);
+
+		NexusObjectWrapper<NXslit> nexusDelegate = new NexusObjectWrapper<>(getName(), slit, NXslit.NX_X_GAP, NXslit.NX_Y_GAP);
+		nexusDelegate.setDefaultAxisDataFieldName(NXslit.NX_X_GAP);
+		nexusDelegate.setCategory(NexusBaseClass.NX_INSTRUMENT);
+		return nexusDelegate;
+	}
+
+	// Class
+
+	private ILazyWriteableDataset xLzSet;
+	private ILazyWriteableDataset yLzSet;
+	private ILazyWriteableDataset xLzValue;
+	private ILazyWriteableDataset yLzValue;
+
+	private boolean writingOn = true;
+
+	public boolean isWritingOn() {
+		return writingOn;
+	}
+
+	public void setWritingOn(boolean writingOn) {
+		this.writingOn = writingOn;
+	}
+
+	@ScanFinally
+	public void nullify() {
+		xLzSet   = null;
+		xLzValue = null;
+		yLzSet   = null;
+		yLzValue = null;
+	}
+
+	/**
+	 * Add the attributes for the given attribute container into the given nexus object.
+	 * @param positioner
+	 * @param container
+	 * @throws NexusException if the attributes could not be added for any reason
+	 */
+	private static void registerAttributes(NXobject nexusObject, IScanAttributeContainer container) throws NexusException {
+		// We create the attributes, if any
+		nexusObject.setField("name", container.getName());
+		if (container.getScanAttributeNames()!=null) {
+			for(String attrName : container.getScanAttributeNames()) {
+				try {
+					nexusObject.setField(attrName, container.getScanAttribute(attrName));
+				} catch (Exception e) {
+					throw new NexusException(MessageFormat.format(
+							"An exception occurred attempting to get the value of the attribute ''{0}'' for the device ''{1}''",
+							container.getName(), attrName), e);
+				}
+			}
+		}
+	}
+
+
+	public String getX_gap_deviceName() {
+		return x_gap_deviceName;
+	}
+
+	public void setX_gap_deviceName(String x_gap_deviceName) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00117)